### ISSUE TITLE
Test case failure on SLES - WebSocketTests/WebSocket_Https_Duplex_TextStreamed 

### ISSUE THREAD

**Issue by StephenBonikowsky** [dotnet/wcf#2561]
Test Case: WebSocket_Https_Duplex_TextStreamed
Located: ...\WCF\src\System.Private.ServiceModel\tests\Scenarios\Extensibility\WebSockets\WebSocketTests.4.1.0.cs

Message
```
System.InvalidOperationException : An error occurred when creating the WebSocket with the factory of type 'CoreClrClientWebSocketFactory'. See the inner exception for details.
---- System.Net.WebSockets.WebSocketException : Unable to connect to the remote server
-------- System.Net.Http.HttpRequestException : The SSL connection could not be established, see inner exception.
------------ System.Security.Authentication.AuthenticationException : The remote certificate is invalid according to the validation procedure.
```
Stack Trace:
```
   at System.ServiceModel.Channels.ClientWebSocketTransportDuplexSessionChannel.CreateWebSocketWithFactoryAsync(X509Certificate2 certificate, TimeoutHelper timeoutHelper) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ClientWebSocketTransportDuplexSessionChannel.cs:line 225
   at System.ServiceModel.Channels.ClientWebSocketTransportDuplexSessionChannel.OnOpenAsync(TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ClientWebSocketTransportDuplexSessionChannel.cs:line 94
   at System.ServiceModel.Channels.CommunicationObject.OnOpenAsyncInternal(TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CommunicationObject.cs:line 461
   at System.ServiceModel.Channels.CommunicationObject.System.ServiceModel.IAsyncCommunicationObject.OpenAsync(TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CommunicationObject.cs:line 436
   at System.ServiceModel.Channels.CommunicationObject.OpenAsyncInternal(TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CommunicationObject.cs:line 410
   at System.ServiceModel.Channels.CommunicationObject.Open(TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CommunicationObject.cs:line 405
   at System.Runtime.TaskHelpers.CallActionAsync[TArg](Action`1 action, TArg argument) in /root/wcf-1374341/src/System.Private.ServiceModel/src/Internals/System/Runtime/TaskHelpers.cs:line 312
   at System.ServiceModel.Channels.CommunicationObject.OpenOtherAsync(ICommunicationObject other, TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CommunicationObject.cs:line 897
   at System.ServiceModel.Channels.ServiceChannel.OnOpenAsync(TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannel.cs:line 1364
   at System.ServiceModel.Channels.CommunicationObject.OnOpenAsyncInternal(TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CommunicationObject.cs:line 461
   at System.ServiceModel.Channels.CommunicationObject.System.ServiceModel.IAsyncCommunicationObject.OpenAsync(TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CommunicationObject.cs:line 436
   at System.ServiceModel.Channels.CommunicationObject.OpenAsyncInternal(TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CommunicationObject.cs:line 410
   at System.ServiceModel.Channels.ServiceChannel.CallOpenOnce.System.ServiceModel.Channels.ServiceChannel.ICallOnce.Call(ServiceChannel channel, TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannel.cs:line 1982
   at System.ServiceModel.Channels.ServiceChannel.CallOnceManager.CallOnce(TimeSpan timeout, CallOnceManager cascade) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannel.cs:line 2073
   at System.ServiceModel.Channels.ServiceChannel.EnsureOpened(TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannel.cs:line 507
   at System.ServiceModel.Channels.ServiceChannel.Call(String action, Boolean oneway, ProxyOperationRuntime operation, Object[] ins, Object[] outs, TimeSpan timeout) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannel.cs:line 764
   at System.ServiceModel.Channels.ServiceChannelProxy.InvokeService(MethodCall methodCall, ProxyOperationRuntime operation) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannelProxy.cs:line 369
   at System.ServiceModel.Channels.ServiceChannelProxy.Invoke(MethodInfo targetMethod, Object[] args) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceChannelProxy.cs:line 138
--- End of stack trace from previous location where exception was thrown ---
   at System.Reflection.DispatchProxyGenerator.Invoke(Object[] args)
   at generatedProxy_3.DownloadStream()
   at WebSocketTests.WebSocket_Https_Duplex_TextStreamed() in /root/wcf-1374341/src/System.Private.ServiceModel/tests/Scenarios/Extensibility/WebSockets/WebSocketTests.4.1.0.cs:line 303
----- Inner Stack Trace -----
   at System.Net.WebSockets.WebSocketHandle.ConnectAsyncCore(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options) in /root/corefx-1373375/src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs:line 184
   at System.Net.WebSockets.ClientWebSocket.ConnectAsyncCore(Uri uri, CancellationToken cancellationToken) in /root/corefx-1373375/src/System.Net.WebSockets.Client/src/System/Net/WebSockets/ClientWebSocket.cs:line 157
   at System.ServiceModel.Channels.CoreClrClientWebSocketFactory.CreateWebSocketAsync(Uri address, WebHeaderCollection headers, ICredentials credentials, WebSocketTransportSettings settings, TimeoutHelper timeoutHelper) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CoreClrClientWebSocketFactory.cs:line 36
   at System.ServiceModel.Channels.ClientWebSocketTransportDuplexSessionChannel.CreateWebSocketWithFactoryAsync(X509Certificate2 certificate, TimeoutHelper timeoutHelper) in /root/wcf-1374341/src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ClientWebSocketTransportDuplexSessionChannel.cs:line 216
----- Inner Stack Trace -----
   at System.Net.Http.ConnectHelper.EstablishSslConnectionAsync(HttpConnectionSettings settings, String host, HttpRequestMessage request, Stream stream, CancellationToken cancellationToken) in /root/corefx-1373375/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs:line 159
   at System.Net.Http.HttpConnectionPool.CreateConnectionAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /root/corefx-1373375/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs:line 197
   at System.Net.Http.HttpConnectionPool.WaitForCreatedConnectionAsync(ValueTask`1 creationTask) in /root/corefx-1373375/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs:line 296
   at System.Net.Http.HttpConnectionPool.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /root/corefx-1373375/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs:line 172
   at System.Net.Http.CookieHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /root/corefx-1373375/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/CookieHandler.cs:line 31
   at System.Net.Http.AuthenticateAndRedirectHandler.SendAsync(HttpRequestMessage request, CancellationToken cancellationToken) in /root/corefx-1373375/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticateAndRedirectHandler.cs:line 53
   at System.Net.WebSockets.WebSocketHandle.ConnectAsyncCore(Uri uri, CancellationToken cancellationToken, ClientWebSocketOptions options) in /root/corefx-1373375/src/System.Net.WebSockets.Client/src/System/Net/WebSockets/WebSocketHandle.Managed.cs:line 116
----- Inner Stack Trace -----
   at System.Net.Security.SslState.StartSendAuthResetSignal(ProtocolToken message, AsyncProtocolRequest asyncRequest, ExceptionDispatchInfo exception) in /root/corefx-1373375/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 973
   at System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest) in /root/corefx-1373375/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 820
   at System.Net.Security.SslState.StartSendBlob(Byte[] incoming, Int32 count, AsyncProtocolRequest asyncRequest) in /root/corefx-1373375/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 791
   at System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest) in /root/corefx-1373375/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 960
   at System.Net.Security.SslState.StartReceiveBlob(Byte[] buffer, AsyncProtocolRequest asyncRequest) in /root/corefx-1373375/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 863
   at System.Net.Security.SslState.CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest) in /root/corefx-1373375/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 820
   at System.Net.Security.SslState.StartSendBlob(Byte[] incoming, Int32 count, AsyncProtocolRequest asyncRequest) in /root/corefx-1373375/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 791
   at System.Net.Security.SslState.ProcessReceivedBlob(Byte[] buffer, Int32 count, AsyncProtocolRequest asyncRequest) in /root/corefx-1373375/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 960
   at System.Net.Security.SslState.PartialFrameCallback(AsyncProtocolRequest asyncRequest) in /root/corefx-1373375/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 1094
--- End of stack trace from previous location where exception was thrown ---
   at System.Net.Security.SslState.EndProcessAuthentication(IAsyncResult result) in /root/corefx-1373375/src/System.Net.Security/src/System/Net/Security/SslState.cs:line 709
   at System.Threading.Tasks.TaskFactory`1.FromAsyncCoreLogic(IAsyncResult iar, Func`2 endFunction, Action`1 endAction, Task`1 promise, Boolean requiresSynchronization)
--- End of stack trace from previous location where exception was thrown ---
   at System.Net.Http.ConnectHelper.EstablishSslConnectionAsync(HttpConnectionSettings settings, String host, HttpRequestMessage request, Stream stream, CancellationToken cancellationToken) in /root/corefx-1373375/src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/ConnectHelper.cs:line 154
```

**Comment by zhenlan** [comment]
This fails only on SLES_12.

**Comment by Lxiamail** [comment]
@StephenBonikowsky Can you please try to run the test to see if the blocking issue has been fixed?

**Comment by StephenBonikowsky** [comment]
Haven't had a chance yet.

**Comment by StephenBonikowsky** [comment]
This is still an issue.
Recently we just determined that a number of tests were being skipped by accident inour VSTS runs, we just got them running again and had 25 test cases failing on the same SLES platform with the same error.

**Comment by StephenBonikowsky** [comment]
This has been fixed with PR #2837 in Master and ported to both release/2.1.0 and release/2.0.0

**Comment by StephenBonikowsky** [comment]
All the skipped test cases are also being re-enabled with PR #2842 and ported to release/2.1.0
Tests were not skipped in release/2.0.0 because we weren't running against SLES.
